src/nupic/utils/Random.cpp

```diff
@@ -37,7 +37,7 @@ bool Random::operator==(const Random &o) const {
 	 gen == o.gen;
 }
 
-bool static_gen_seeded = false;
+bool static_gen_seeded = false;  //TODO avoid the static variables?
```

Comment:
This is a global variable, and it's **not** in the header file so it only assessable from this file.

I think a better way to organize this would be to make this into a private static member of the Random class.